File: java/openjdk17_spring/src/main/java/benchmark/openjdk17_spring/Customer.java
package benchmark.openjdk17_spring;

import org.bson.codecs.pojo.annotations.BsonProperty;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("customers")
public record Customer(
        String guid,
        @BsonProperty("first_name")
        String firstName,
        @BsonProperty("last_name")
        String lastName,
        String email,
        String address
){}

File: java/kotlin_spring/src/main/kotlin/benchmark/kotlin_spring/Customer.kt
package benchmark.kotlin_spring

import org.bson.codecs.pojo.annotations.BsonProperty
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document("customers")
data class Customer(

        val guid: String?,
        val first_name: String?,
        val last_name: String?,
        val email: String?,
        val address: String?
)